packages/contracts-bedrock/src/L1/OptimismPortal.sol

```diff
@@ -404,11 +413,15 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
                 }
             }
 
-            // Make a call to the target contract only if there is calldata.
-            if (_tx.data.length != 0) {
-                success = SafeCall.callWithMinGas(_tx.target, _tx.gasLimit, 0, _tx.data);
+            if (_tx.value != 0 && l2Sender == Predeploys.L2_STANDARD_BRIDGE) {
+                success = SafeCall.callWithMinGas(_tx.target, _tx.gasLimit, _tx.value, _tx.data);
             } else {
-                success = true;
+                // Make a call to the target contract only if there is calldata.
+                if (_tx.data.length != 0) {
+                    success = SafeCall.callWithMinGas(_tx.target, _tx.gasLimit, 0, _tx.data);
+                } else {
+                    success = true;
+                }
             }
```

Comment:
I guess it's better to avoid nested conditions for more readability and decrease the size of the contract
```
if (_tx.value != 0 && l2Sender == Predeploys.L2_STANDARD_BRIDGE) {
    success = SafeCall.callWithMinGas(_tx.target, _tx.gasLimit, _tx.value, _tx.data);
} else if (_tx.data.length != 0) {
    success = SafeCall.callWithMinGas(_tx.target, _tx.gasLimit, 0, _tx.data);
} else {
    success = true;
}
```

Reply:
Fixed